File: queue_using_stacks.py
class Queue(object):
    def __init__(self):
        self.name = 'S->13'
        self.s1 = []
        self.s2 = []

    def enqueue(self, value):
        self.s1.append(value)

    def dequeue(self):
        if self.s2:
            return self.s2.pop()
        if self.s1:
            while self.s1:
                self.s2.append(self.s1.pop())
            return self.s2.pop()
        return None
q = Queue()
q.enqueue(1)
q.enqueue(2)
q.enqueue(3)
print(q.dequeue())
q.enqueue(4)
q.enqueue(5)
print(q.dequeue())
print(q.dequeue())
print(q.dequeue())
print(q.dequeue())

File: Python/queue_using_stacks.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <stack>
using namespace std;
class Queue {
public:
    Queue(){}
    void Enqueue(int data) {
        s1.push(data);
    }
    
    void Dequeue() {
        if (s2.empty()) {
            while(!s1.empty()) {
                s2.push(s1.top());
                s1.pop();
            }
        }
        s2.pop();
    }
    
    int Front() {
        if (s2.empty()) {
            while(!s1.empty()) {
                s2.push(s1.top());
                s1.pop();
            }
        }
        return s2.top();
    } 
private:
    stack<int> s1;
    stack<int> s2;
};

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */ 
    Queue q;
    int qr;
    int command;
    cin >> qr;
    for (int i = 0; i < qr; ++i) {
        cin >> command;
        if (command == 1) {
            int num;
            cin >> num;
            q.Enqueue(num);
        }
        else if (command == 2) {
            q.Dequeue();
        }
        else if (command == 3) {
            cout << q.Front() << endl;
        }
    }
    
    return 0;
}
